### ISSUE TITLE
Should the `prototype` keyword be marked unstable?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#23026]
The `prototype` keyword and `module` concept never seemed to gain a lot of popularity.  Now that we have attributes, it seems like something we might specify with an attribute on the module rather than a keyword.  To that end, should we mark `prototype` as unstable in order to reclaim the keyword and not commit to it for 2.0?

**Comment by bradcray** [comment]
[With a quick check, this is the main reserved word in the spec that jumped out at me as being questionable for 2.0, which is encouraging, assuming the spec is in sync with the parser].

**Comment by mppf** [comment]
I'd prefer to have an attribute for this.

**Comment by lydia-duncan** [comment]
It looks as though the discussion on #20856 concluded with removing the 2.0 label because the third error mode involved using a pragma and pragmas are already unstable.  But I think we got distracted from the original question of if we should use an attribute instead of `prototype` as a keyword.  Since we're past 2.0, the answer is probably "no".  But it might be interesting to log this as "3.0" instead?  I wouldn't immediately make a 3.0, but marking it as such would mean we are more likely to think about it again when we feel as though we have enough things we'd like to adjust to warrant it

**Comment by bradcray** [comment]
I think this is the first thing I've felt regretful we didn't get done before 2.0.  Not to say there won't be other/bigger things, but it'd be a shame if we were stuck with this keyword forever over this.  :)  And I'm glad that we made it almost two weeks before I had that reaction to something.  :D 

If we had a process for making breaking changes between 2.0 and 3.0 (e.g., establishing an "invested Chapel users" board of voting members who could agree "This is obviously not something we want in Chapel"), I'd put this on the agenda and would be surprised if anyone advocated for keeping it (or was even using it, potentially).

If we don't establish such a process, I'd probably design and implement the @attribute as soon as possible and create a `--[no-]warn-chapel-3.0` flag, potentially turning it on by default, that would steer uses of `prototype` to the attribute as being the preferred style going forward.

**Comment by mppf** [comment]
> If we don't establish such a process, I'd probably design and implement the `@Attribute` as soon as possible and create a --[no-]warn-chapel-3.0 flag, potentially turning it on by default, that would steer uses of prototype to the attribute as being the preferred style going forward.

Agreed: IMO the starting point is to implement an attribute alternative. From there, we can add an on-by-default warning requesting that people use the attribute, even without it being a breaking change. Then we can deprecate and remove `prototype` in a 3.0 or potentially before (if we develop a process for it).
